### DISCUSSION TITLE
How to access the local file system  for Theia browser frontend that is connecting with remote Theia backend

### DISCUSSION THREAD

**Discussion by mbilal100** [eclipse-theia/theia#10842] [Q&A]
Theia frontend is connecting with Theia browser backend which is running on separate remote machine. Now I can access the remote file system (where beckend is running) from Theia frontend but how can I access the local file system where frontend in running??

**Comment by msujew** [comment]
Hi @mbilal100,

it depends on how much access do you need to the local user file system. If you need only specific files, you can simply ask users to upload them to the remote workspace, using the `Upload files...` context menu on the explorer.

If you want to run everything on the local file system, everything is a bit more involved. You would need to create a file system provider that is based on the [local file system API](https://developer.mozilla.org/en-US/docs/Web/API/File_System_Access_API) provided by browsers. Be aware that VSCode extensions in the backend cease to function, see [VSCode web extensions](https://code.visualstudio.com/api/extension-guides/web-extensions).

**Reply by mbilal100** [reply]
Hi @msujew,
Thank you for reply,

> it depends on how much access do you need to the local user file system. If you need only specific files, you can simply ask users to upload them to the remote workspace, using the Upload files... context menu on the explorer.

Thanks it works.

> If you want to run everything on the local file system, everything is a bit more involved. You would need to create a file system provider that is based on the [local file system API](https://developer.mozilla.org/en-US/docs/Web/API/File_System_Access_API) provided by browsers. Be aware that VSCode extensions in the backend cease to function, see [VSCode web extensions](https://code.visualstudio.com/api/extension-guides/web-extensions).

Do we have any extension that support this filesystem API or any Theia menu option for this?


Dose Theia also supports VScode remote https://code.visualstudio.com/docs/remote/remote-overview ?
